integration/src/main/java/scripts/testscripts/ControlledGcsBucketLifecycle.java

```diff
@@ -193,11 +194,21 @@ public void doUserJourney(TestUserSpecification testUser, WorkspaceApi workspace
     logger.info("Added {} as a reader to workspace {}", reader.userEmail, getWorkspaceId());
 
     // TODO(PF-643): this should happen inside WSM.
-    logger.info("Waiting 30s for permissions to propagate");
-    TimeUnit.SECONDS.sleep(30);
-
-    // Second user can now read the blob
-    Blob readerRetrievedFile = readerStorageClient.get(blobId);
+    int numTries = 20;
```

Comment:
Can you pull this into a utility or something similar? We have a few other tests which use a similar pattern of "add user to workspace, wait for cloud permissions" that could reuse this logic